tabular/src/autogluon/tabular/models/tabular_nn/tabular_nn_model.py

```diff
@@ -343,7 +343,12 @@ def train_net(self, train_dataset, params, val_dataset=None, initialize=True, se
             if val_dataset is not None:
                 val_metric = self.score(X=val_dataset, y=y_val, metric=self.stopping_metric)
                 if (val_metric >= best_val_metric) or (e == 0):
-                    if not np.isnan(val_metric):
+                    if np.isnan(val_metric):
```

Comment:
This new code block should go above line 345, if we ever get NaN it should break.